### DISCUSSION TITLE
Does adonisjs provide supports for method overriding?

### DISCUSSION THREAD

**Discussion by c4710n** [adonisjs/core#2069] [Help] [answered]
Almost all fullstack frameworks provide this feature, or it will difficult to update an resource via HTML form when following RESTful rules.

+ (ruby) Rails supports it
+ (elixir) Phoenix supports it
+ (nodejs) Express supports [it](http://expressjs.com/en/resources/middleware/method-override.html)

Does adonisjs support it?
> I can't find any related information in the docs.

**Comment by c4710n** [comment, marked as answer]
Found [this line](https://github.com/adonisjs/http-server/blob/396a35375e8211500982f277fe26abc598781731/src/Request/index.ts#L325) in source code.

Hope it will help others.

**Comment by RomainLanz** [comment]
Hey @c4710n! 👋 

What you are looking for is called `Method Spoofing`.
We haven't yet documented this in the V5 documentation (it's available on [V4 website](https://adonisjs.com/docs/4.1/request#_method_spoofing)).

If you are using AdonisJS 5 and you want to activate method spoofing, you need to modify the settings inside the `config/app.ts` file.

```ts
// ...

export const http: HttpConfig = {
  /*
  |--------------------------------------------------------------------------
  | Allow method spoofing
  |--------------------------------------------------------------------------
  |
  | Method spoofing enables defining custom HTTP methods using a query string
  | `_method`. This is usually required when you are making traditional
  | form requests and wants to use HTTP verbs like `PUT`, `DELETE` and
  | so on.
  |
  */
  allowMethodSpoofing: false,

  // ...
```
